dlib/python/numpy_image.h

```diff
@@ -70,11 +73,9 @@ class numpy_rgb_image
     numpy_rgb_image (boost::python::object& img) 
     {
         long shape[3];
-        get_numpy_ndarray_parts(img, _data, shape);
+        get_numpy_ndarray_parts(img, _data, _contig_buf, shape);
         _nr = shape[0];
         _nc = shape[1];
-        if (shape[2] != 3)
```

Comment:
I compare `pixel_traits<T>::num` to the last dimension only when it is greater than 1. So it should work with grayscale image as well.